pennylane/math/quantum.py

```diff
@@ -1125,3 +1125,88 @@ def _compute_max_entropy(density_matrix, base):
     maximum_entropy = qml.math.log(rank) / div_base
 
     return maximum_entropy
+
+
+def min_entropy(state, indices, base=None, check_state=False, c_dtype="complex128"):
+    r"""Compute the minimum entropy from a state vector or density matrix on a given subsystem. It supports all
+    interfaces (Numpy, Autograd, Torch, Tensorflow and Jax).
+
+    .. math::
+        S_{\text{max}}( \rho ) = \log( \text{rank} ( \rho ))
+
+    Args:
+        state (tensor_like): ``(2**N)`` state vector or ``(2**N, 2**N)`` density matrix.
+        indices (list(int)): List of indices in the considered subsystem.
+        base (float): Base for the logarithm. If None, the natural logarithm is used.
+        check_state (bool): If True, the function will check the state validity (shape and norm).
+        c_dtype (str): Complex floating point precision type.
+
+    Returns:
+        float: The minimum entropy of the considered subsystem.
+
+    **Example**
+
+    The minimum entropy of a subsystem for any state vector can be obtained. Here is an example for the
+    maximally entangled state, where the subsystem entropy is maximal (default base for log is exponential).
+
+    >>> x = [1, 0, 0, 1] / np.sqrt(2)
+    >>> min_entropy(x, indices=[0])
+    -0.6931472
+
+    The logarithm base can be changed. For example:
+
+    >>> min_entropy(x, indices=[0], base=2)
+    -1.0
+
+    The minimum entropy can be obtained by providing a quantum state as a density matrix. For example:
+
+    >>> y = [[1/2, 0, 0, 1/2], [0, 0, 0, 0], [0, 0, 0, 0], [1/2, 0, 0, 1/2]]
+    >>> min_entropy(y, indices=[0])
+    -0.6931472
+
+    The Von Neumann entropy is always greater than the minimum entropy.
+    >>> x = [np.cos(np.pi/8), 0, 0, -1j*np.sin(np.pi/8)]
+    >>> vn_entropy(x, indices=[1])
+    0.4164955
+    >>> min_entropy(x, indices=[1])
+    -0.15834718382037496
+
+    """
+    density_matrix = reduced_dm(state, indices, check_state, c_dtype)
+    minimum_entropy = _compute_min_entropy(density_matrix, base)
+
+    return minimum_entropy

```

Comment:
```suggestion
    return _compute_min_entropy(density_matrix, base)
```

This isn't in pylint or black, but some other linters recommend "inline variable that is immediately returned". This is just a personal preference thing, so feel free to ignore the comment.